### PR TITLE
Implement 'use-problem-json' rule

### PR DESCRIPTION
**Pull request by s0k0**

### CODE REVIEW

File: src/main/java/de/zalando/zally/rules/UseProblemJsonRule.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @Override
+    public List<Violation> validate(Swagger swagger) {
+        List<Violation> violations = new ArrayList<>();
+        if (swagger == null || swagger.getPaths() == null) {
+            return violations;
+        }
+        for(Map.Entry<String, Path> path: swagger.getPaths().entrySet())
+            for(Map.Entry<HttpMethod, Operation> op : path.getValue().getOperationMap().entrySet())
+                for(Map.Entry<String, Response> response: op.getValue().getResponses().entrySet())
+                    try {
```
Comment (netme) [inline src/main/java/de/zalando/zally/rules/UseProblemJsonRule.java]: Is it possible to minimise the scope of `try..catch` block? Probably the following scope should be good enough:

```java
try {
     Integer httpCodeInt = Integer.valueOf(response.getKey());
} catch(NumberFormatException nfe) {
}
```

File: src/main/java/de/zalando/zally/rules/UseProblemJsonRule.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                        Integer httpCodeInt = Integer.valueOf(response.getKey());
+                        //TODO: check if schema reference type is Problem := https://github.com/swagger-api/swagger-parser/pull/303/files
+                        if (httpCodeInt >= 400 && httpCodeInt < 600 && response.getValue().getSchema() != null
```
Comment (netme) [inline src/main/java/de/zalando/zally/rules/UseProblemJsonRule.java]: Optional: `<= 599` will be easy to understand. Took me a while to understand why 600 :-)

Comment (s0k0) [inline src/main/java/de/zalando/zally/rules/UseProblemJsonRule.java]: Done.

File: src/main/java/de/zalando/zally/rules/UseProblemJsonRule.java (hunk 1 above)
Comment (s0k0) [inline src/main/java/de/zalando/zally/rules/UseProblemJsonRule.java]: Agreed, I edited the scope.

**Comment by ishishov** [comment]
👍

**Pull request merged**